Toolchains/toolchain-generic.cmake

```diff
@@ -0,0 +1,57 @@
+# This is tuned for Ubuntu multiarch / multilib + clang + lld
+
+if ("${GUEST_ARCH}" STREQUAL "x86_32")
+    set(CMAKE_SYSTEM_PROCESSOR i386)
+
+    # Fun Fact: Ubuntu's i386 actually uses the i686 compiler
+    set(triple i386-linux-gnu)
+    set(sysroot_triple i686-linux-gnu)
+elseif ("${GUEST_ARCH}" STREQUAL "x86_64")
+    set(CMAKE_SYSTEM_PROCESSOR x86_64)
+
+    # the triple identifies the target platform
+    set(triple x86_64-linux-gnu)
+    set(sysroot_triple x86_64-linux-gnu)
+else()
+    message(FATAL_ERROR "Toolchain doesn't support guest arch: |${GUEST_ARCH}|")
+endif()
+
+# Force re-checking of the compiler - cmake sometimes gets confused here and skips it
+set(CMAKE_C_ABI_COMPILED False)
+set(CMAKE_CXX_ABI_COMPILED False)
+
+
+# By default we use the clang & lld that was used to compile and link FEX, in cross compiler mode
+#
+set(CMAKE_C_COMPILER clang)
+set(CMAKE_CXX_COMPILER clang++)
+# we need to use lld for cross compiling
+add_link_options("-fuse-ld=lld")
+
+
+# gcc alternative
+#
+## set(CMAKE_C_COMPILER i686-linux-gnu-gcc)
+## set(CMAKE_C_FLAGS -m32)
+## set(CMAKE_CXX_COMPILER i686-linux-gnu-g++)
+## set(CMAKE_CXX_FLAGS -m32)
+
+set(CMAKE_C_COMPILER_TARGET ${triple})
+set(CMAKE_CXX_COMPILER_TARGET ${triple})
+
+# Set the compiler to looks for libs, includes and packages in the correct folders
+# as seen on the official cmake site
```

Comment:
Can you provide a specific reference? I couldn't find documentation on the official cmake site that seemed relevant to what we're doing here.

Reply:
https://cmake.org/cmake/help/latest/variable/CMAKE_SYSROOT_COMPILE.html

Reply:
The defaults in cross compiling are actually empty, it seems, so you need to set it up. Sadly, random partially working examples and this documentation is all that the cmake team provides - google is your friend to find out more info.

I don't have the exact reference handy, it was like 2-3 google searches away when I found it and copy pasted it.

Reply:
In general, I think the only 'predictable' way to support this is having a full dev rootfs, as we have discussed in the meets, which kind of falls outside the scope of this PR. @Sonicadvance1 said they will look into it when they have iirc.